util/patch/patch.go

```diff
@@ -277,7 +293,7 @@ func (h *Helper) shouldPatch(in string) bool {
 
 // calculate changes tries to build a patch from the before/after objects we have
 // and store in a map which top-level fields (e.g. `metadata`, `spec`, `status`, etc.) have changed.
-func (h *Helper) calculateChanges(after client.Object) (map[string]bool, error) {
+func (h *Helper) calculateChanges(after client.Object) (map[string]interface{}, error) {
```

Comment:
nit: this func both sets a field in the struct and returns a value, which is not immediate to notice.
Can we return both patchDiff and changes?

Reply:
Done - alternatively we could get rid of the changes element and replace it with the patchDiff and modify the shouldPatch function. This would simplify the code a bit I think, but I didn't want to change the implementaton of the patcher too much initially, which is why I set changes internally instead.